### DISCUSSION TITLE
Specify complete VersaTiles stack

### DISCUSSION THREAD

**Discussion by MichaelKreil** [versatiles-org/versatiles-documentation#18] [Ideas]
I have compiled all my experiences with VersaTiles and expanded the specification. So far, we have only specified the container. The extension should also define the complete stack including minimum requirements for the generator, server, network and frontend.
What do you think? /cc @yetzt 
https://github.com/versatiles-org/versatiles-spec/blob/v02/v02/readme.md

**Comment by yetzt** [comment]
To me it is unclear who the audience for this document is. Is it meant as "this is the state of implementation we want to adhere to" or is it "if you want to build a compatible stack, this is what you need to implement"? I believe we are in need of some sort of roadmap, i don't know if this document is the right fit for that particular need.

> Vector tiles must conform to the [Shortbread schema](https://shortbread-tiles.org/).

This is s a bit moot if we want to provide vector tile sets beyond the scope of OSM/Shortbread, for example vectors of elevation contour lines. I believe what you meant is, "versatiles provides a vector tile layer that is always in the OSM/Shortbread schema", but it reads as "only this one schema is allowed".

> Mandatory HTTP service provision.

The meaning of this is unclear to me. Does it mean "Versatiles Server must always provide an HTTP Interface over a TCP port?" 

The server layer should also recommend graceful downgrade of compression and encoding, eg deliver gzip or png/jpeg respectively if the client can't handle brotli/webp.

Otherwise it makes good sense to me.

**Reply by MichaelKreil** [reply]
Good points.
